### PR TITLE
replace intreface with more SDK like interface

### PR DESCRIPTION
**Pull request by Tieske**
When updating the docs and reviewing the official SDK implementation, I changed the interface to be more like the AWS SDK for JS.

Please have a look. The example code explains how to use it after the changes.

WARNING: I didn't have an RDS database to test against, so please verify it still works
(I did verify the local config changes to work)

### CODE REVIEW

File: src/resty/aws/service/rds/signer.lua [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    getAuthToken = getAuthToken,  -- injected method for token generation
+  }
+
+  -- first copy the inherited config elements NOTE: inherits from AWS, not the rds_instance!!!
+  for k,v in pairs(rds_instance.aws.config) do
+    token_instance.config[k] = v
+  end
+
+  -- service specifics
+  -- see https://github.com/aws/aws-sdk-js/blob/9295e45fdcda93b62f8c1819e924cdb4fb378199/lib/rds/signer.js#L11-L15
```
Comment (windmgc) [inline src/resty/aws/service/rds/signer.lua:124]: Interesting...so the JS SDK consider this as part of the RDS "service class"

Comment (Tieske) [reply src/resty/aws/service/rds/signer.lua:124]: exactly

**Comment by windmgc** [comment]
@Tieske  I fixed the param(from self to self.config) that passed to presign function and fixed test accordingly. After the fix I tested locally with my RDS and it worked as expected, thanks for the interface refactor!

File: src/resty/aws/service/rds/signer.lua [hunk 2]
```diff
@@ -99,7 +99,7 @@ local function getAuthToken(self, opts) --endpoint, region, db_user)
     },
   }
 
-  local presigned_request, err = presign_awsv4_request(self, req_data, opts.signingName, region, RDS_IAM_AUTH_EXPIRE_TIME)
+  local presigned_request, err = presign_awsv4_request(self.config, req_data, opts.signingName, region, RDS_IAM_AUTH_EXPIRE_TIME)
```
Comment (Tieske) [inline src/resty/aws/service/rds/signer.lua:102]: yup, sorry about that. 👍

**Pull request merged**